src/Microsoft.Health.Dicom.Functions/Registration/ReplaySafeCounter.cs

```diff
@@ -0,0 +1,34 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics.Metrics;
+using EnsureThat;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+namespace Microsoft.Health.Dicom.Functions.Registration;
+
+public class ReplaySafeCounter
```

Comment:
It's unfortunate there's no interface/class that can be used to make our counter look like the underlying counter. Perhaps a comment here to stay that the Counter class is sealed and other Counter methods can be added as necessary